AirCasting/ABConnector/SDCardAirBeamServices.swift

```diff
@@ -153,8 +160,33 @@ class BluetoothSDCardAirBeamServices: SDCardAirBeamServices {
     private func finishSync(completion: () -> Void) {
         self.bluetoothManager.unsubscribeCharacteristicObserver(self.dataCharacteristicObserver!)
         self.bluetoothManager.unsubscribeCharacteristicObserver(self.metadataCharacteristicObserver!)
+        expectedMeasurementsCount = [:]
+        receivedMeasurementsCount = [:]
+        if let timerToken = timerToken {
+            timer.stop(token: timerToken)
+        }
         completion()
     }
+    
+    private func startMonitoringForEnd(completion: @escaping () -> Void) -> Void {
+        var checkedMeasurementsCount: [SDCardSessionType: Int] = [:]
+        timerToken = timer.schedule(every: 1) { [weak self] in
```

Comment:
we're using a lot of time-based stuff in our core logic components, we shouldn't be using regular main-runloop bound timers. It can cause hard to track bugs. Good thing we're using abstraction (here, but not in other places)

Something to put on a todo list.